src/axom/slic/tests/slic_macros.cpp

```diff
@@ -122,6 +122,21 @@ TEST(slic_macros, test_error_macros)
   SLIC_ERROR_IF(false, "this message should not be logged!");
   EXPECT_TRUE(slic::internal::is_stream_empty());
 
+  // Check selective filtering based on root == false
```

Comment:
Is this test run in parallel?
I think we'd want it to be to ensure we don't deadlock.

Reply:
This is just setting all nodes to have the root value of false.  It should not deadlock.  It is also not a parallel test.

Reply:
Sorry, I should have been clearer.
I think we should have a parallel test where some ranks are `root` and others are not. That test should make sure that calling `SLIC_ERROR_ROOT_IF` does not deadlock.

Reply:
You just want everything now don't you...

Reply:
added in newest commit.